core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -197,8 +197,6 @@ package object config {
 
   private[spark] val EVENT_LOG_ROLLING_MAX_FILES_TO_RETAIN =
     ConfigBuilder("spark.eventLog.rolling.maxFilesToRetain")
```

Comment:
Hmm... this is kind of a history server config, right? Or even more specifically, a `FsHistoryProvider` config.

Might be better to have it grouped with other configs in `History.scala`, and make it a constructor argument in `EventLogFileCompactor`.

(In fact the same applies to `EVENT_LOG_COMPACTION_SCORE_THRESHOLD`. Should that be not `internal()` anymore, also?)

Reply:
> Hmm... this is kind of a history server config, right? Or even more specifically, a FsHistoryProvider config.
> Might be better to have it grouped with other configs in History.scala, and make it a constructor argument in EventLogFileCompactor.

Yeah... the name and place was constructed when I wasn't fully sure where to put, so I just put near neighborhood configuration. But I think it makes sense to move to History, and also add `history` prefix into configuration name to indicate end users that it won't run at driver side. Making it as a part of constructor argument would be also good.

> (In fact the same applies to EVENT_LOG_COMPACTION_SCORE_THRESHOLD. Should that be not internal() anymore, also?)

I'm not 100 % sure who would tune the score manually. Original intention was to make it flexible to ease testing so placed it to configuration and set it to be "internal". If we want to expose this configuration, we should maybe also provide the information on score and the statistics so that end users can see and tune the threshold. WDYT?

Reply:
Makes sense to keep the score threshold internal. We can make it public later if there's a good reason for it.